src/box/sql/select.c

```diff
@@ -5290,6 +5290,10 @@ sqlSelectPrep(Parse * pParse,	/* The parser context */
 	sql *db;
 	if (NEVER(p == 0))
 		return;
+	// Nothing to do if we are already bound
+	// for cached AST
+	if (p->selFlags & SF_CachedAST)
```

Comment:
OOPS! These are remnants of older code which tried to cache AST structures inside of prepared queries cache. This was cleaned up recently (because it was fragile), but this flag remains